src/test/rustdoc-ui/intra-link-errors.rs

```diff
@@ -2,7 +2,16 @@
 //~^ NOTE lint level is defined
 
 // FIXME: this should say that it was skipped (maybe an allowed by default lint?)
-/// [<invalid syntax>]
+/// [invalid intra-doc syntax!!]
+
+// These will be resolved as regular links.
+/// [this is <invalid syntax>](https://www.rust-lang.org)
+/// [this is <invalid syntax>]
+/// [<invalid syntax>](https://www.rust-lang.org)
+/// [<invalid syntax>][rlo]
+///
+/// [this is <invalid syntax>]: https://www.rust-lang.org
+/// [rlo]: https://www.rust-lang.org
```

Comment:
These should go in `src/test/rustdoc` so you can check the links actually resolved to the right place.

Reply:
Indeed that caught a bug! It stripped the generics even if it decided that it wasn't an intra-doc link.